### PR TITLE
Main PR for 2022-2023

### PR DESCRIPTION
**Pull request by raghavauppuluri13**
integration pr for 2022-2023

### CODE REVIEW

File: lunabot_embedded/src/teensy_driver_node.cpp
Comment (raghavauppuluri13) [inline lunabot_embedded/src/teensy_driver_node.cpp:58]: Make more clear on what publish does (update function name?)

File: lunabot_control/scripts/manual_controller.py
Comment (ashviniyer21) [inline lunabot_control/scripts/manual_controller.py:30]: Constrain between -1 and 1 before multiplying by 127 (just for safety)

File: lunabot_msgs/msg/RobotState.msg
Comment (ashviniyer21) [inline lunabot_msgs/msg/RobotState.msg:1]: Rename to be more clear (robot sensors?) State makes it seem like it should store position after SLAM

**Review by ashviniyer21 — APPROVED** [review]
Added couple of minor comments, overall looks good to merge

File: lunabot_embedded/firmware/update_board_manager.py
Comment (ashviniyer21) [inline lunabot_embedded/firmware/update_board_manager.py:8]: if __name__ == "__main__" this code? Didn't look at all files but same feedback if this exists in other places

**Comment by raghavauppuluri13** [comment]
> LGTM. Left 1 minor comment (not even sure if that's your code or if it's library code lol). Didn't look in too much detail but trusting feedback from `feature/autonomy` PR was addressed properly.

fixed! will merge then.

**Comment by raghavauppuluri13** [comment]
> LGTM. Left 1 minor comment (not even sure if that's your code or if it's library code lol). Didn't look in too much detail but trusting feedback from `feature/autonomy` PR was addressed properly.

for clarity, these were the ones not addressed in this PR (non-blocking):

higher priority fixes

https://github.com/PurdueLunabotics/purdue_lunabotics/pull/43#discussion_r1451099217

nits + docs updates

https://github.com/PurdueLunabotics/purdue_lunabotics/pull/43#discussion_r1421110396

https://github.com/PurdueLunabotics/purdue_lunabotics/pull/43#discussion_r1264551454

https://github.com/PurdueLunabotics/purdue_lunabotics/pull/43#discussion_r1264551573

https://github.com/PurdueLunabotics/purdue_lunabotics/pull/43#discussion_r1451519852

https://github.com/PurdueLunabotics/purdue_lunabotics/pull/43#discussion_r1264553342

File: readme.md
Comment (Isaac-F24) [inline readme.md:1]: https://wiki.purduearc.com/wiki/tutorials/setup-ros is a dead link.

Comment (raghavauppuluri13) [reply readme.md:1]: good catch! should be fixed now

**Review by ashviniyer21 — CHANGES_REQUESTED** [review]
Blocking PR merge until changes are tested on the robot

**Comment by raghavauppuluri13** [comment]
blocked by https://github.com/PurdueLunabotics/purdue_lunabotics/pull/55. once that's tested in sim and real, can merge this.

File: readme.md
Comment (Isaac-F24) [inline readme.md:35]: Suggestion: add recurse submodules instructions

File: lunabot_embedded/firmware/update_board_manager.py
Comment (raghavauppuluri13) [reply lunabot_embedded/firmware/update_board_manager.py]: done

File: lunabot_msgs/msg/RobotState.msg
Comment (raghavauppuluri13) [reply lunabot_msgs/msg/RobotState.msg:1]: done

File: readme.md
Comment (raghavauppuluri13) [reply readme.md]: done

**Pull request merged**
